echopype/consolidate/api.py

```diff
@@ -55,84 +60,123 @@ def swap_dims_channel_frequency(ds: Union[xr.Dataset, str, pathlib.Path]) -> xr.
+    downward : bool, default `True`
+        The transducers point downward.
+    use_platform_vertical_offsets: bool, default `False`
+        If True, use Echodata Platform group vertical offset values to compute transducer depth.
+        Currently only implemented for EK60/EK80 sonar models.
+    use_platform_angles: bool, `False`
+        If True, use Echodata Platform group angle values to compute `echo_range` scaling values.
+        Currently only implemented for EK60/EK80 sonar models.
+        Cannot be used in tandem with Echodata Beam group angles.
+    use_beam_angles: bool `False`
+        If True, use Echodata Beam group angle values to compute `echo_range` scaling values.
+        Currently only implemented for EK60/EK80 sonar models.
+        Cannot be used in tandem with Echodata Platform group angles.
 
     Returns
     -------
-    The input dataset with a `depth` variable (in meters) added
+    The input dataset with a `depth` variable (in meters) added.
+    """
+    # Open Sv dataset
+    ds = open_source(ds, "dataset", {})
 
-    Notes
-    -----
-    Currently this function only scalar inputs of depth_offset and tilt angle.
-    In future expansion we plan to add the following options:
+    # Raise `ValueError` if `echodata` is needed but not passed in
+    if (not echodata) and (use_platform_vertical_offsets or use_platform_angles or use_beam_angles):
+        raise ValueError(
+            "If any of `use_platform_vertical_offsets` or `use_platform_angles` is `True` "
+            + "then `echodata` cannot be `None`."
+        )
 
-    * Allow inputs as xr.DataArray for time-varying variations of these variables
-    * Use data stored in the EchoData object or raw-converted file from which the Sv is derived,
-      specifically `water_level`, `vertical_offtset` and `tilt` in the `Platform` group.
-    """
-    # TODO: add options to use water_depth, vertical_offset, tilt stored in EchoData
-    # # Water level has to come from somewhere
-    # if depth_offset is None:
-    #     if "water_level" in ds:
-    #         depth_offset = ds["water_level"]
-    #     else:
-    #         raise ValueError(
-    #             "water_level not found in dataset and needs to be supplied by the user"
-    #         )
-
-    # # If not vertical needs to have tilt
-    # if not vertical:
-    #     if tilt is None:
-    #         if "tilt" in ds:
-    #             tilt = ds["tilt"]
-    #         else:
-    #             raise ValueError(
-    #                 "tilt not found in dataset and needs to be supplied by the user. "
-    #                 "Required when vertical=False"
-    #             )
-    # else:
-    #     tilt = 0
+    # Raise `NotImplementedError` if `use_platform_angles` and `use_beam_angles` are
+    # both true.
+    if use_platform_angles and use_beam_angles:
+        raise NotImplementedError(
+            "Computing depth with both platform and beam angles is not implemented yet."
+        )
```

Comment:
I think we should also add an explicit check for `tilt` here if either `use_platform_angles` or `use_beam_angles` are `True` when `tilt` is also given. Also see my comment above about `tilt` taking precedence over `use_platform_angles` or `use_beam_angles`.

Reply:
I agree 👍